docs/vcpkg-schema-definitions.schema.json

```diff
@@ -0,0 +1,715 @@
+            },
+            "commit": {
+              "type": "string",
+              "description": "The commit SHA or ref that will be checked out."
+            },
+            "options": {
+              "type": "array",
+              "items": {
+                "type": "string",
+                "enum": [
+                  "full",
+                  "recurse"
+                ]
+              }
+            },
+            "subdirectory": {
+              "type": "string",
+              "default": ""
+            }
+          },
+          "required": [
+            "git"
+          ],
+          "additionalProperties": false
+        }
+      ]
+    },
+    "git-registry": {
+      "type": "object",
+      "description": "A port registry backed by a git repo.",
+      "properties": {
+        "kind": {
+          "type": "string",
+          "pattern": "git"
+        },
+        "repository": {
+          "type": "string",
+          "description": "The URI or SSH entry to clone from. (as if by `git clone`)"
+        },
+        "baseline": {
+          "description": "The SHA that will be checked out to find default versions for ports from this registry.",
+          "$ref": "#/definitions/sha1"
+        },
+        "ref": {
+          "type": "string",
+          "description": "The ref to fetch when cloning this git registry."
+        }
+      },
+      "required": [
+        "kind",
+        "repository",
+        "baseline"
+      ]
+    },
+    "has-schemed-version": {
+      "type": "object",
+      "oneOf": [
+        {
+          "properties": {
+            "version-string": {
+              "description": "Text used to identify an arbitrary version",
+              "type": "string",
+              "pattern": "^[^#]+$"
+            }
+          },
+          "required": [
+            "version-string"
+          ]
+        },
+        {
+          "properties": {
+            "version": {
+              "description": "A relaxed version string (1.2.3.4...)",
+              "type": "string",
+              "pattern": "^\\d+(\\.\\d+)*(?:-((?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\\.(?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\\+([0-9a-zA-Z-]+(?:\\.[0-9a-zA-Z-]+)*))?$"
+            }
+          },
+          "required": [
+            "version"
+          ]
+        },
+        {
+          "properties": {
+            "version-semver": {
+              "$ref": "#/definitions/semantic-version"
+            }
+          },
+          "required": [
+            "version-semver"
+          ]
+        },
+        {
+          "properties": {
+            "version-date": {
+              "description": "A date version string (e.g. 2020-01-20)",
+              "type": "string",
+              "pattern": "^\\d{4}-\\d{2}-\\d{2}(\\.\\d+)*$"
+            }
+          },
+          "required": [
+            "version-date"
+          ]
+        }
+      ]
+    },
+    "identifier": {
```

Comment:
It would be nice to have a dedicated `"feature-name"` definition with a better description.

Reply:
I think making better descriptions for all the things would be a great future improvement